pkg/config/node.go

```diff
@@ -191,32 +195,104 @@ func GetVRRPConfig(apiVip, ingressVip net.IP) (vipIface net.Interface, nonVipAdd
 	return getInterfaceAndNonVIPAddr(vips)
 }
 
-func IsUpgradeStillRunning(kubeconfigPath string) (error, bool) {
+// GetNodes will collect all nodes which contain the annotation node-role.kubernetes.io/${VAR} annotation
+// and return a map with the data.
+//
+// i.e:
+//	node-role.kubernetes.io/master
+//	node-role.kubernetes.io/worker
+//	node-role.kubernetes.io/infra
+//	etc.
+//
+// Args:
+//      - kubeconfigPath as string
+//
+// Returns:
+//      - map[string][]v1.Node or error
+//
+func GetNodes(kubeconfigPath string) (map[string][]v1.Node, error) {
+
+	nodeCluster := make(map[string][]v1.Node)
 
 	config, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
 	if err != nil {
-		return err, true
+		return nil, err
 	}
 
 	clientset, err := kubernetes.NewForConfig(config)
 	if err != nil {
-		return err, true
+		return nil, err
 	}
 
 	nodes, err := clientset.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
 	if err != nil {
-		return err, true
+		return nil, err
 	}
 
 	for _, node := range nodes.Items {
-		if node.Annotations["machineconfiguration.openshift.io/desiredConfig"] != node.Annotations["machineconfiguration.openshift.io/currentConfig"] ||
-			node.Annotations["machineconfiguration.openshift.io/desiredConfig"] != nodes.Items[0].Annotations["machineconfiguration.openshift.io/desiredConfig"] {
+		// Get all labels for *each node* which contain node-role.kubernetes.io/
+		for k, _ := range node.Labels {
```

Comment:
nit: rename `k` to something like `label`. In addition `, _` should not be needed in this case.